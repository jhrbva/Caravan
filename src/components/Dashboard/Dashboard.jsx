import React from 'react';

import './Dashboard.scss';
import { Col, Row } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import SummaryCard from '../SummaryCard/SummaryCard';
import BigButton from '../BigButton/BigButton';
import MailOutlinedIcon from '@material-ui/icons/MailOutlined';
import CardTravelIcon from '@material-ui/icons/CardTravel';
import AirportShuttleIcon from '@material-ui/icons/AirportShuttle';

export default class Dashboard extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			invitations: [],
			tripsJoined: [],
			tripsHosted: [],
		};
	}

	componentDidMount() {
		Promise.all([
			fetch(`/trips/${this.props.userid}`),
			fetch(`/invitations/${this.props.userid}`),
		])
			.then(([dataTrips, dataInvitations]) => {
				return Promise.all([dataTrips.json(), dataInvitations.json()]);
			})
			.then(([dataTrips, dataInvitations]) => {
				this.setState({
					tripsHosted: dataTrips.tripsHosted,
					tripsJoined: dataTrips.tripsJoined,
					invitations: dataInvitations,
				});
			});
	}

	renderSection = (title, type, icon) => {
		const isYourTrips = title === 'Your Trips' ? true : false;
		const nullResponse =
			title === 'Invitations' ? 'No invitations' : 'No trips';

		return (
			<>
				<h2>{title}</h2>
				<Row>
					{type.length ? (
						type.map((entry, id) => {
							return (
								<Col md={4}>
									<SummaryCard
										key={id}
										trip={entry}
										icon={icon}
										host={entry.hostname}
										members={entry.members}
										reststops={entry.reststops}
										isYourTrips={isYourTrips}
									/>
								</Col>
							);
						})
					) : (
						<p>{nullResponse}</p>
					)}
				</Row>
			</>
		);
	};

	render() {
		if (this.props.userid === 0) {
			return (
				<>
					<p>
						Hi there! <br /> Login first to see your trips
					</p>
					<Link to='/'>
						<BigButton value={'Login'} />
					</Link>
				</>
			);
		} else {
			const { invitations, tripsJoined, tripsHosted } = this.state;
			return (
				<>
					<div className='dashboard-wrapper'>
						<Link to='/trip'>
							<BigButton value='+ New Trip' color={'green'} />
						</Link>
						<span></span>
						<Link to='/ec'>
							<BigButton value='Update Emergency Contact' color={'blue'} />
						</Link>

						<div className='trip-section'>
							{this.renderSection(
								'Invitations',
								invitations,
								<MailOutlinedIcon fontSize='large' />
							)}
						</div>
						<div className='trip-section'>
							{this.renderSection(
								'Your Trips',
								tripsHosted,
								<CardTravelIcon fontSize='large' />
							)}
						</div>
						<div className='trip-section'>
							{this.renderSection(
								'Trips Joined',
								tripsJoined,
								<AirportShuttleIcon fontSize='large' />
							)}
						</div>
					</div>
				</>
			);
		}
	}
}
