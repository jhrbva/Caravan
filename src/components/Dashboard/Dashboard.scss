@import '../../utilities/colorPalette.scss';
@import '../../utilities/mediaQueries.scss';
