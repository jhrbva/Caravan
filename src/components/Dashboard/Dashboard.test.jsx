import React from 'react';
import { shallow } from 'enzyme';
import { Dashboard } from './Dashboard';

const mockTrip = {
	trip_title: 'Brunch at gold statue :P',
	trip_description: "Let's see Rockefeller",
	startlocation: '1307 Florence Avenue, Plainfield, NJ 07060',
	destination: '45 Rockefeller Plaza, New York, NY, 10111',
	tripdate: '2020-06-01T13:55:06.000Z',
};
const reststops = [{ location: 'Wawa' }, { location: 'Loves' }];
const host = {username: 'conniecode'};
const members = [{username: 'jcode'}];

describe('Dashboard', () => {
	it('should match the snapshot', () => {
		const wrapper = shallow(<Dashboard user={{}} />);
		wrapper.setState({ invitations: [mockTrip] });
		wrapper.setState({ tripsJoined: [mockTrip] });
		wrapper.setState({ tripsHosted: [mockTrip] });
		wrapper.setState({ host: [host] });
		wrapper.setState({ members: [members] });
		wrapper.setState({ reststops: [reststops] });

		expect(wrapper).toMatchSnapshot();
	});
});
