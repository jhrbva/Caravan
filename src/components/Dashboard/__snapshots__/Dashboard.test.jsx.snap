// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should match the snapshot 1`] = `
<Fragment>
  <h1>
    Dashboard
  </h1>
  <Link
    to="/trip"
  >
    <Button
      active={false}
      disabled={false}
      type="button"
      variant="success"
    >
      New Trip +
    </Button>
  </Link>
  <h3>
    Invitations
  </h3>
  <Row
    noGutters={false}
  >
    <Col>
      <SummaryCard
        host={Array []}
        key="0"
        reststops={Array []}
        trip={
          Object {
            "destination": "45 Rockefeller Plaza, New York, NY, 10111",
            "startlocation": "1307 Florence Avenue, Plainfield, NJ 07060",
            "trip_description": "Let's see Rockefeller",
            "trip_title": "Brunch at gold statue :P",
            "tripdate": "2020-06-01T13:55:06.000Z",
          }
        }
      />
    </Col>
  </Row>
  <h3>
    Your Trips
  </h3>
  <Row
    noGutters={false}
  >
    <Col>
      <SummaryCard
        host={Array []}
        key="0"
        reststops={Array []}
        trip={
          Object {
            "destination": "45 Rockefeller Plaza, New York, NY, 10111",
            "startlocation": "1307 Florence Avenue, Plainfield, NJ 07060",
            "trip_description": "Let's see Rockefeller",
            "trip_title": "Brunch at gold statue :P",
            "tripdate": "2020-06-01T13:55:06.000Z",
          }
        }
      />
    </Col>
  </Row>
  <h3>
    Trips Joined
  </h3>
  <Row
    noGutters={false}
  >
    <Col>
      <SummaryCard
        host={Array []}
        key="0"
        reststops={Array []}
        trip={
          Object {
            "destination": "45 Rockefeller Plaza, New York, NY, 10111",
            "startlocation": "1307 Florence Avenue, Plainfield, NJ 07060",
            "trip_description": "Let's see Rockefeller",
            "trip_title": "Brunch at gold statue :P",
            "tripdate": "2020-06-01T13:55:06.000Z",
          }
        }
      />
    </Col>
  </Row>
</Fragment>
`;
